network.py

```diff
@@ -641,25 +604,40 @@ def _post_request(self, url, headers, body, token=None,
 
         return ret
 
-    @staticmethod
-    def access_url(conn, session, method, full_url, headers, data,
-                   proxies, timeout=(
-                    DEFAULT_CONNECT_TIMEOUT,
-                    DEFAULT_CONNECT_TIMEOUT,
-                    DEFAULT_REQUEST_TIMEOUT),
-                   token=None,
-                   is_raw_text=False,
-                   catch_okta_unauthorized_error=False,
-                   is_raw_binary=False,
-                   is_raw_binary_iterator=True,
-                   use_ijson=False, is_single_thread=False):
-        logger = getLogger(__name__)
-
-        connection_timeout = timeout[0:2]
-        request_timeout = timeout[2]  # total request timeout
-        request_thread_timeout = 60  # one request thread timeout
-
-        def request_thread(result_queue):
+    def fetch(self, method, full_url, headers, *, data=None, timeout=None, **kwargs):
```

Comment:
Interesting, I was pretty sure I tested that particular signature on 2.7, but I do recall that keyword only arguments handling changed in py3k.  (Disclaimer: I rarely use python2)